pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4219,6 +4222,46 @@ int uprobe_seq_ops_trigger(struct pt_regs *ctx)
     return 0;
 }
 
+SEC("uprobe/trigger_mem_dump_event")
+int uprobe_mem_dump_trigger(struct pt_regs *ctx)
+{
+    u64 address = 0;
+    u64 size = 0;
+    u32 trigger_pid = bpf_get_current_pid_tgid() >> 32;
+
+#if defined(bpf_target_x86)
+    address = ctx->bx; // 1st arg
+    size = ctx->cx;    // 2nd arg
+#elif defined(bpf_target_arm64)
+    address = ctx->user_regs.regs[1]; // 1st arg
+    size = ctx->user_regs.regs[2];    // 2nd arg
+#else
+    return 0;
+#endif
+
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    // uprobe was triggered from other tracee instance
+    if (data.config->tracee_pid != trigger_pid)
+        return 0;
+
+    if (size <= 0)
+        return 0;
+
+    int ret = save_bytes_to_buf(&data, (void *) address, size & MAX_MEM_DUMP_SIZE, 1);
+    // return in case of failed pointer read
+    if (ret == 0) {
+        tracee_log(ctx, BPF_LOG_LVL_ERROR, BPF_LOG_ID_MEM_READ, ret);
+        return 0;
+    }
+    save_to_submit_buf(&data, (void *) &address, sizeof(void *), 0);
```

Comment:
Putting index 0 after index 1 will reverse the order of these args in the table printer.
Better put it in the correct order (or change the order in the event definition)